src/Illuminate/Database/Query/JsonExpression.php

```diff
@@ -43,7 +43,7 @@ protected function getJsonBindingParameter($value)
                 return '?';
             case 'object':
             case 'array':
-                return '?';
+                return 'json_extract(?, "$")';
```

Comment:
Shorter alternative:
```suggestion
                return 'cast(? as json)';
```

Reply:
Yes, that's better.